src/metadataGeneration/resolveType.ts

```diff
@@ -20,7 +20,7 @@ type UsableDeclaration = ts.InterfaceDeclaration
   | ts.ClassDeclaration
   | ts.TypeAliasDeclaration;
 
-export function ResolveType(typeNode: ts.TypeNode): Type {
+export function ResolveType(typeNode: ts.TypeNode, depth = 1): Type {
```

Comment:
It might be desired to modify this function further, but I have zero experience with typescript reflection and culdn't find any documentation on that yet.